tests/e2e/framework/testInfo.go

```diff
@@ -111,6 +123,102 @@ func (t testInfo) Update(r int) error {
 	return t.createStatusFile(r)
 }
 
+func (t testInfo) FetchAndSaveClusterLogs() error {
+	if *projectID == "" {
+		return nil
+	}
+	// connect to stackdriver
+	ctx := context.Background()
+	glog.Info("Fetching cluster logs")
+	loggingClient, err := logging.NewClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	fetchAndWrite := func(logName string) error {
+		glog.Info(fmt.Sprintf("Fetching logs on %s\n", logName))
+		// fetch logs from pods created for this run only
+		filter := fmt.Sprintf(
+			`logName = "%s" AND
+			resource.labels.namespace_id = "%s"`,
+			logName, *namespace)
+		req := &loggingpb.ListLogEntriesRequest{
+			ResourceNames: []string{"projects/" + *projectID},
+			Filter:        filter,
+		}
+		it := loggingClient.ListLogEntries(ctx, req)
+		// create log file in append mode
+		prefix := fmt.Sprintf("projects/%s/logs/", *projectID)
+		logID := logName[len(prefix):]
+		path := filepath.Join(t.LogsPath, fmt.Sprintf("%s.log", logID))
+		f, err := os.OpenFile(path, os.O_WRONLY|os.O_APPEND|os.O_CREATE, 0600)
+		if err != nil {
+			return err
+		}
+		// fetch log entries with pagination
+		var entries []*loggingpb.LogEntry
+		pager := iterator.NewPager(it, pageSize, "")
+		for page := 0; ; page++ {
+			pageToken, err := pager.NextPage(&entries)
+			if err != nil {
+				glog.Errorf("Iterator paging failed: %v", err)
+				return err
+			}
+			// append logs to file
+			for _, logEntry := range entries {
+				_, err = f.WriteString(fmt.Sprintf("%v\n", logEntry.Payload))
+				if err != nil {
+					return err
+				}
+			}
+			if pageToken == "" {
+				break
+			}
+		}
+		return f.Close()
+	}
+
+	req := &loggingpb.ListLogsRequest{
+		Parent: "projects/" + *projectID,
+	}
+	it := loggingClient.ListLogs(ctx, req)
+	errMsg := ""
+	for {
+		logName, err := it.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			return err
+		}
+		if i := strings.Index(logName, "test-infra-presubmit"); i == -1 {
+			if err := fetchAndWrite(logName); err != nil {
+				errMsg += err.Error() + "\n"
+			}
+		}
+	}
+
+	for _, resrc := range resources {
+		glog.Info(fmt.Sprintf("Fetching deployment info on %s\n", resrc))
+		path := filepath.Join(t.LogsPath, fmt.Sprintf("%s.yaml", resrc))
+		if yaml, err0 := util.Shell(fmt.Sprintf("kubectl get %s -o yaml", resrc)); err0 == nil {
```

Comment:
If you take a look at line 141 to 144, the namespace is passed in from commend line and is used to filter/select log entries of that namespace only. 

Also inverted the if branches so that conditions always check for err != nil. I agree that doing so improves readability.